### PR TITLE
[common/function] refactor: simplify get_layout_offsets

### PR DESCRIPTION
**Pull request by drmingdrmer**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Changelog

- Improvement

## Related Issues

### CODE REVIEW

File: common/functions/src/aggregates/aggregate_function_state.rs [hunk 1]
```diff
@@ -100,25 +100,21 @@ impl From<StateAddr> for usize {
...
+    for func in funcs {
+        let layout = func.state_layout();
+        let align = layout.align();
+
+        total_size = (total_size + align - 1) / align * align;
+
+        offsets.push(total_size);
+
+        max_align = max_align.max(align);
+        total_size += layout.size();
```
Comment (sundy-li) [inline common/functions/src/aggregates/aggregate_function_state.rs:116]: 「size: 2, align: 8」

total_size will be `3`, max_align will be `8` ?

Comment (drmingdrmer) [reply common/functions/src/aggregates/aggregate_function_state.rs:116]: Hmmm? the result should be `total_size=2, max_align=8`?

That's just what the former version does in a simpler way.

Comment (drmingdrmer) [reply common/functions/src/aggregates/aggregate_function_state.rs:116]: Did you mean that the size should not be smaller than align or something else?

Comment (sundy-li) [reply common/functions/src/aggregates/aggregate_function_state.rs:116]: > Hmmm? the result should be total_size=2, max_align=8? That's just what the former version does in a simpler way.

Aha, you are right, I was wrong.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**